js/valid.js

```diff
@@ -0,0 +1,463 @@
+	function valid()
+	{
+	    try {
+	    	var first_name = document.getElementById('fname').value;
+			var middle_name = document.getElementById('mname').value;
+		    var last_name = document.getElementById('lname').value;
+		    var dob = document.getElementById('dob').value;
+		    var email = document.getElementById('email').value;
+		    var atpos = email.indexOf("@");
+		    var dotpos = email.lastIndexOf(".");
+		    var  password = document.getElementById('password').value;
+		    var confirm_password = document.getElementById('confirm_password').value;
+		    var employer = document.getElementById('employer').value;
+		    //var image =  document.getElementById('image').value;
+		    var home_street = document.getElementById('home_street').value;
+		    var home_city = document.getElementById('home_city').value;
+		    var home_state = document.getElementById('home_state').value;
+		    var home_mobile = document.getElementById('home_mobile').value;
+		    var home_zip = document.getElementById('home_zip').value;
+		    var home_landline = document.getElementById('home_landline').value;
+		    var home_fax = document.getElementById('home_fax').value;
+		    var office_street = document.getElementById('office_street').value;
+		    var office_state = document.getElementById('office_state').value;
+		    var office_city = document.getElementById('office_city').value;
+		    var office_zip = document.getElementById('office_zip').value;
+		    var office_mobile = document.getElementById('office_mobile').value;
+		    var office_landline = document.getElementById('office_landline').value;
+		    var office_fax = document.getElementById('office_fax').value;
+		    var communication = document.getElementsByName('communication[]');
+		    var flag = 0;
+
+	    	var f_mess, m_mess, l_mess, dob_mess, email_mess, password_mess, confirm_password_mess, employer_mess,
+	    	home_street_mess, home_state_mess, home_city_mess, home_mobile_mess,home_zip_mess, home_landline_mess, home_fax_mess,
+	    	office_street_mess, office_city_mess, office_state_mess, office_zip_mess, office_mobile_mess, office_landline_mess,
+	    	office_fax_mess, comm_mess, login_mess; 
+
+		    var letters = /^[A-Za-z]+$/;
+		    var number =  /^\d{10}$/;
+
+		    if(first_name == null || first_name == ' ')
+		    {	
+		    	f_mess = "Please fill the field";
+		    	//alert(mess);
+		    }
+		    else if(first_name.match(letters))
+		    {	
+		    	f_mess = ' ';
+		    }
+		    else
+		    {
+		    	
+		    	f_mess = " wrong input";
+		    }
+
+		    /*------------------------------------------------*/
+		    if(middle_name == null || middle_name == ' ' || middle_name == '')
```

Comment:
trim function can be used here instead of 3 OR conditions. Please note, with OR condition, the program has to check for every conditional statement.